configure.in

```diff
@@ -72,7 +72,11 @@ if test "X$NGX_OPENSSL_SRC" != X ; then
   NGX_OPENSSL_INSTALL_DIR=$NGX_OPENSSL_SRC/.openssl
   NGX_CFLAGS=-I$NGX_OPENSSL_INSTALL_DIR/include
   if test "x$enable_dynamic_module" = xyes ; then
-      NGX_LDFLAGS="-L$NGX_OPENSSL_INSTALL_DIR/lib\ -Wl,-rpath=$NGX_OPENSSL_INSTALL_DIR/lib"
+    if [[ "$OSTYPE" == "darwin"* ]]; then
+        NGX_LDFLAGS="-L$NGX_OPENSSL_INSTALL_DIR/lib\ -Wl,-rpath,$NGX_OPENSSL_INSTALL_DIR/lib"
```

Comment:
I guess it's a Clang issue, not macOS issue.
Gcc accepts "-Wl,-rpath,/path/to/dir" command line option. So I think we can replace "-rpath=" with "-rpath," if it works popular Linux distros.

Reply:
@yyamano I see. 
That's mean I just need to use `-rpath,` without condition right?

Reply:
@Zogoo I think so.

Reply:
@yyamano , @matsumotory 
I have changed it to as check compiler. Because, I think this update came from latest pull request. So, I guess someone using Clang for development.

Reply:
@Zogoo How does it work? configure is a bourne shell script. I think # is interpreted as a comment line. bash specific syntax?

https://github.com/Zogoo/ngx_mruby/blob/build_with_openssl/configure#L3710-L3714

~~~
#if defined(__clang__)
      NGX_LDFLAGS="-L$NGX_OPENSSL_INSTALL_DIR/lib\ -Wl,-rpath,$NGX_OPENSSL_INSTALL_DIR/lib"
#elif defined(__GNUC__) || defined(__GNUG__)
      NGX_LDFLAGS="-L$NGX_OPENSSL_INSTALL_DIR/lib\ -Wl,-rpath=$NGX_OPENSSL_INSTALL_DIR/lib"
#endif
~~~

Reply:
> I think this update came from latest pull request. 

What do you mean? Could you explain what's the update?

Reply:
I think "-Wl,-rpath,$NGX_OPENSSL_INSTALL_DIR/lib" works on almost all platforms.
See https://lekensteyn.nl/rpath.html and https://gcc.gnu.org/ml/gcc-help/2005-12/msg00017.html

I confirmed it works on NetBSD 8.0 (gcc 5.5.0), macOS 10.14 (clang 11.0) and CentOS 7.6 (gcc 4.8.5).

Reply:
Sorry for the delay. 
I think I missed with my last commit, I was trying to check compiler with some google result but seems like it won't work.
Thanks for your clarification, let me change it as your way.

Reply:
@Zogoo Looks good to me.

Reply:
@yyamano  thanks for your help